gfx/video_shader_parse.c

```diff
@@ -1121,6 +1118,27 @@ void video_shader_write_conf_cgp(config_file_t *conf,
    }
 }
 
+const char *video_shader_to_str(enum rarch_shader_type type)
+{
+   switch (type)
+   {
+      case RARCH_SHADER_CG:
+         return "Cg";
+      case RARCH_SHADER_HLSL:
+         return "HLSL";
+      case RARCH_SHADER_GLSL:
+         return "GLSL";
+      case RARCH_SHADER_SLANG:
+         return "Slang";
+      case RARCH_SHADER_METAL:
+         return "Metal";
+      case RARCH_SHADER_NONE:
+         return "none";
+      default:
+         return "???";
+   }
```

Comment:
Same here. A function returning a const char * should always return something, so just change this back to something where "default" would do a break; first, then return "???" at the end.